File: platforms/android/runtime/core/src/main/java/io/nativeblocks/runtime/api/provider/wandkit/Wandkit.kt
package io.nativeblocks.runtime.api.provider.wandkit

import android.content.Context
import io.nativeblocks.runtime.api.NativeblocksEdition

/**
 * Defines the contract for managing the WandKit functionality within the native framework.
 */
interface Wandkit {

    /**
     * Sets up the WandKit with the specified context and edition.
     * @param context The Android context required for initialization.
     * @param edition The edition of Nativeblocks being used.
     * @param instanceName The name of Nativeblocks instance being used.
     */
    fun setup(context: Context, edition: NativeblocksEdition, instanceName: String)

    /**
     * Cleans up and releases any resources used by the WandKit.
     */
    fun destroy()
}

File: platforms/ios/runtime/Sources/NativeblocksRuntime/api/provider/wandkit/Wandkit.swift
import Foundation

/// Protocol that defines the core behavior for managing Wandkit operations.
/// The `Wandkit` protocol provides methods to set up and destroy the Wandkit instance, making it suitable for managing the lifecycle of Nativeblocks components.
public protocol Wandkit {
    /// Sets up the Wandkit instance with the specified edition.
    /// - Parameter edition: The `NativeblocksEdition` that determines the configuration type, such as cloud or community.
    func setup(edition: NativeblocksEdition, instanceName: String)

    /// Destroys the Wandkit instance and releases all associated resources.
    func destroy()
}
